### PR TITLE
List method for SubjectsController

### PR DESCRIPTION
**Pull request by VeronikaRevjakina**
#346 

- [x]  Method

- [x] TestListSuccess

- [x] Add ACL

- [x]  testListNotFound

- [x] testListForbidden

### CODE REVIEW

**Comment by brarrow** [comment]
@VeronikaRevjakina can you please complete your PR? I need your SubjectsMapper. Maybe i can somehow help you with it?

**Comment by brarrow** [comment]
[checkstyle.zip](https://github.com/malinink/easy-tests/files/2019444/checkstyle.zip)
Try this. Now it gonna be allright. And you also can install checkstyle plugin in ide. In the wiki there information about how to do it.

**Comment by VeronikaRevjakina** [comment]
@malinink check this PR as fast as you can ,please,because greenbarrow needs Subjects mapper (which I've changed). So there will be conflicts if we change mapper at the same time.

File: src/main/java/easytests/api/v1/controllers/SubjectsController.java [hunk 1]
```diff
@@ -1,34 +1,63 @@
...
+        final UserModelInterface userModel = this.usersService.find(userId);
+
+        if (userModel == null) {
+            throw new NotFoundException();
+        }
+        if (!this.acl.hasAccess(userModel)) {
+            throw new ForbiddenException();
+        }
+
+        final List<SubjectModelInterface> subjectsModels = this.subjectsService.findByUser(new UserModelEmpty(userId));
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/SubjectsController.java]: Uh, use userModel you have declared in 45 line

File: src/main/java/easytests/api/v1/mappers/SubjectsMapper.java [hunk 2]
```diff
@@ -9,5 +12,15 @@
  */
 @Service("SubjectsMapperV1")
 public class SubjectsMapper extends ModelMapper {
+    public SubjectsMapper() {
+        super();
+        this.createTypeMap(SubjectModel.class, Subject.class)
+            .addMappings(
+                mapper -> mapper.when(
```
Comment (malinink) [inline src/main/java/easytests/api/v1/mappers/SubjectsMapper.java]: do not copy paste code, that you don't understand. When here is not necessary, remove.

**Review by malinink — CHANGES_REQUESTED** [review]

**Comment by VeronikaRevjakina** [comment]
@malinink improved mistakes,check please

File: src/main/java/easytests/api/v1/controllers/SubjectsController.java [hunk 3]
```diff
@@ -1,34 +1,62 @@
 package easytests.api.v1.controllers;
 
+import easytests.api.v1.exceptions.ForbiddenException;
 import easytests.api.v1.mappers.SubjectsMapper;
+import easytests.api.v1.models.Subject;
+import easytests.common.exceptions.NotFoundException;
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/SubjectsController.java]: use v1 NotFound exception

**Review by malinink — CHANGES_REQUESTED** [review]

**Review by malinink — APPROVED** [review]

**Pull request merged**